src/test/abilities/aroma_veil.test.ts

```diff
@@ -0,0 +1,62 @@
+import { Moves } from "#app/enums/moves";
+import { Species } from "#app/enums/species";
+import { Abilities } from "#enums/abilities";
+import GameManager from "#test/utils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
+import { BattlerTagType } from "#app/enums/battler-tag-type";
+import { ArenaTagType } from "#app/enums/arena-tag-type";
+import { BattlerIndex } from "#app/battle";
+
+describe("Moves - Aroma Veil", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .battleType("double")
+      .enemyAbility(Abilities.BALL_FETCH)
+      .enemyMoveset([Moves.HEAL_BLOCK, Moves.IMPRISON, Moves.SPLASH])
+      .enemySpecies(Species.SHUCKLE)
+      .ability(Abilities.AROMA_VEIL)
+      .moveset([Moves.GROWL]);
+  });
+
+  it("Aroma Veil protects the Pokemon's side against most Move Restriction Battler Tags", async () => {
+    await game.classicMode.startBattle([Species.REGIELEKI, Species.BULBASAUR]);
+
+    const playerPokemon = game.scene.getParty()!;
+
+    game.move.select(Moves.GROWL);
+    game.move.select(Moves.GROWL);
+    await game.forceEnemyMove(Moves.HEAL_BLOCK);
+    await game.toNextTurn();
+    expect(playerPokemon[0].getTag(BattlerTagType.HEAL_BLOCK)).toBeUndefined();
+    expect(playerPokemon[1].getTag(BattlerTagType.HEAL_BLOCK)).toBeUndefined();
+  });
+
+  it("Aroma Veil does not protect against Imprison", async () => {
+    await game.classicMode.startBattle([Species.REGIELEKI, Species.BULBASAUR]);
+
+    const playerPokemon = game.scene.getParty()!;
+
+    game.move.select(Moves.GROWL);
+    game.move.select(Moves.GROWL);
```

Comment:
```suggestion
    game.move.select(Moves.GROWL, 1);
```
Technically you need to specify when it's the second Pokémon (though for the moment this doesn't matter because all player Pokémon are forced to have the same moveset when the moveset override is used).